### ISSUE TITLE
Qt6 Porting

### ISSUE THREAD

**Issue by letsfindaway** [letsfindaway/OpenBoard#67]
Sooner or later a port to Qt6 will be required. Here I describe a strategy to do that:

- [x] Avoid use of deprecated Qt5 APIs. Here already some work exists, which is however not merged. See PR 211, where this will most probably be handled. Would be nice if this could be merged with 1.6.2. **Note:** merged to `1.7-dev`
- [x] Setup a development environment with Qt6, either Leap 15.4 (probably does not include Qt6 right away) or TW. **Note:** Qt6.2 is available for Leap 15.3, but `quazip1-qt6` will come with Leap 15.4.
- [x] Refactor code where necessary, but maintain compatibility with Qt5.

From what I have heard Qt6 porting should be easy.

**Comment by letsfindaway** [comment]
## List of necessary changes

- [ ] `QRegExp` -> `QRegularExpression`
- [ ] `QMediaObject` -> `QMediaPlayer`
- [ ] `QStringRef` -> `QStringView`
- [ ] svgwidget cleanup of `#include`s

... to be extended

**Comment by JBBgameich** [comment]
Pulling some things(like the buildsystem) out of that branch would probably make sense. But I don't think it's possible any more to actually merge it at this point. It was created when I wanted to test all my individual changes in one branch, so it's not necessarily well separated. However it does have a few cool things, for example with that branch I could generate working windows installers using [Craft](https://community.kde.org/Craft) which allows to replace the whole OpenBoard-ThirdParty repository with a proper package manager.

**Comment by JBBgameich** [comment]
The Craft stuff is here btw in case it ever becomes useful: https://invent.kde.org/jbbgameich/craft-blueprints-kde/-/commits/work/openboard/

**Comment by kaamui** [comment]
Hi,

I just tested your refactor-qt6 branch and was able to compile OpenBoard with Qt 6.4. However, I had to remove pkg-config usage, because for some reason it was not working.

I'm not familiar with it so it may just be that I didn't make some configuration. Can you explain me how it is supposed to work ?

**Comment by kaamui** [comment]
> * The installed version has to provide pkgconfig files. This is not so clear. On openSUSE they do not install such files, because of the issues with Qt 6.2. QuaZip does not care whether it runs with Qt 6.2 or Qt 6.4, so it would be binary compatible. But because Qt 6.2 does not provide pkgconfig files, openSUSE decided to remove them from QuaZip-Qt6.

I think it's the issue there for me

**Comment by letsfindaway** [comment]
> You know what I think of all of this. If it tends to make build from source harder for anyone willing to try to build OpenBoard himself, I'm not a big fan.. I lost quite a a few hours where it worked instantly with the common include declaration and linking...
> 
> I'm not sure to see what's the plus-value here

If cmake is properly set up, then it makes building for everyone easier. However it is not always easy to setup. I still see the value.

**Comment by kaamui** [comment]
My worry is that it may become harder form anyone to build OpenBoard on their own because of that. Cmake can be hard to understand until you're used to it. 

And I though pkg-config would be simpler but it was hard for me to make it work. I finally didn't figure it out and had to use simple include and linking methods instead. At this point, I won't integrate it in the main branch. It's not a priority for me, and I want the project to remain simple. 

If you don't mind, until we manage to have something working perfectly, I would be glad if you seperate those commits from the refactor-qt6 branch, as I plan to merge it. Moreover, it has nothing to do with qt6 refactoring

**Comment by letsfindaway** [comment]
As I understand now, `pkgconfig` is not the way to go for Qt 6, due to the problems already discussed. So I will in parallel further work on a CMake build system, which is fortunately completely unrelated to the `qmake` build and just an addition. And I hope I will have something working perfectly in the future ;)

**Comment by letsfindaway** [comment]
merged
